### PR TITLE
Function to get total grad norm of parameters

### PR DESCRIPTION
**Pull request by ranamihir**
Currently, there's no inbuilt function for getting _just_ the Frobenius norm of the parameter gradients. `torch.nn.utils.clip_grad_norm_` returns the desired value, but (obviously) simultaneously clips the gradients in-place.

This PR creates a generic function `clip_grad_and_get_total_norm_` for getting the norm, and optionally clipping the gradients if param `max_norm` is provided. Using this function, `torch.nn.utils.clip_grad_norm_` would effectively remain unchanged, and an additional function `total_grad_norm()` is created, which simply returns the desired norm.

### CODE REVIEW

**Comment by ngimel** [comment]
Passing infinity as max_norm would achieve what you want (parameters not change, value of the parameter norm returned).

**Comment by ranamihir** [comment]
Agreed @ngimel -- but I still felt it's a hacky way to get the norms, and also not immediately obvious enough that the values won't change to those who don't bother to look at the source code.

**Comment by ngimel** [comment]
So it warrants maybe documentation change to make it more explicit that current APIs can be used to do what you want, but not adding new untested function. If you want to actually add new public function, the process is to open issue with a suggestion, get an ok from maintainers and then do that.

**Comment by ranamihir** [comment]
Thanks. WIll do.
